python-avd/pyavd/_eos_designs/structured_config/overlay/router_bgp.py

```diff
@@ -592,43 +598,43 @@ def _neighbors(self: AvdStructuredConfigOverlay) -> list | None:
-                        overlay_peering_interface=self.shared_utils.vtep_loopback,
-                    )
-                    neighbors.append(neighbor)
+        if self.shared_utils.is_wan_client:
+            if not self._ip_in_listen_ranges(self.shared_utils.vtep_ip, self.shared_utils.wan_listen_ranges):
+                msg = f"{self.shared_utils.vtep_loopback} IP {self.shared_utils.vtep_ip} is not in the Route Reflector listen range prefixes"
+                raise AristaAvdError(msg)
+            for wan_route_server in self.shared_utils.filtered_wan_route_servers:
+                neighbor = self._create_neighbor(
+                    wan_route_server.vtep_ip,
+                    wan_route_server.hostname,
+                    self.inputs.bgp_peer_groups.wan_overlay_peers.name,
+                    overlay_peering_interface=self.shared_utils.vtep_loopback,
+                )
+                neighbors.append(neighbor)
 
-                if self.shared_utils.wan_ha:
-                    neighbor = {
-                        "ip_address": self._wan_ha_peer_vtep_ip(),
-                        "peer": self.shared_utils.wan_ha_peer,
-                        "description": self.shared_utils.wan_ha_peer,
-                        "remote_as": self.shared_utils.bgp_as,
-                        "update_source": "Dps1",
-                        "route_reflector_client": True,
-                        "send_community": "all",
-                        "route_map_in": "RM-WAN-HA-PEER-IN",
-                        "route_map_out": "RM-WAN-HA-PEER-OUT",
-                    }
-                    neighbors.append(neighbor)
+            if self.shared_utils.wan_ha:
+                neighbor = {
+                    "ip_address": self._wan_ha_peer_vtep_ip(),
+                    "peer": self.shared_utils.wan_ha_peer,
+                    "description": self.shared_utils.wan_ha_peer,
+                    "remote_as": self.shared_utils.bgp_as,
+                    "update_source": "Dps1",
+                    "route_reflector_client": True,
+                    "send_community": "all",
+                    "route_map_in": "RM-WAN-HA-PEER-IN",
+                    "route_map_out": "RM-WAN-HA-PEER-OUT",
+                }
+                neighbors.append(neighbor)
 
-            if self.shared_utils.is_wan_server:
-                # No neighbor configured on the `wan_overlay_peers` peer group as it is covered by listen ranges
-                for wan_route_server in self.shared_utils.filtered_wan_route_servers:
-                    neighbor = self._create_neighbor(
-                        wan_route_server.vtep_ip,
-                        wan_route_server.hostname,
-                        self.inputs.bgp_peer_groups.wan_rr_overlay_peers.name,
-                        overlay_peering_interface=self.shared_utils.vtep_loopback,
-                    )
-                    neighbors.append(neighbor)
+        if self.shared_utils.is_wan_server:
```

Comment:
```suggestion
        elif self.shared_utils.is_wan_server:
```

Reply:
hmm this is the same as we cannot be both client and server but chganged it

Reply:
just code optimization